### DISCUSSION TITLE
Optical flow discrete "steps"/staircase effect

### DISCUSSION THREAD

**Discussion by manuelmaiorb** [carla-simulator/carla#5165] [Q&A]
Hello! For some time I have been trying to obtain some optical flow image frames from a simulation I run using Carla. I have spawned an Optical Flow sensor and process the images using a function which looks like this:
```python
def _parse_image_optical_flow_wide(image_raw):
    array_optical_flow = np.frombuffer(image_raw.raw_data, dtype=np.float32)
    array_optical_flow = np.reshape(array_optical_flow, (image_raw.height, image_raw.width, 2))
    array_optical_flow = np.dstack((array_optical_flow, np.zeros((image_raw.height, image_raw.width, 1))))
    array_optical_flow = array_optical_flow[:, :, ::-1]
    array_optical_flow[:, :, 1] *=  0.5
    array_optical_flow[:, :, 2] *= 0.5
    array_optical_flow = array_optical_flow[12:524, 0:1664]
    cv2.imwrite('out/OpticalFlowWide/OpticalFlowWide_' + str(CameraManager.count_images_optical_flow_wide) + '.exr', array_optical_flow.astype("float32"))
    CameraManager.count_images_optical_flow_wide += 1
```

**Requirements**
- HDR optical flow image frames (16 or 32 bits per channel, floating point | red _for X_, blue _for Y_, green _not used_). 
- Optical Flow data expressed in pixel units

**Problem**
The problem is that the output image frames have a weird discrete steps/staircase effects:
![optical_flow](https://user-images.githubusercontent.com/92081210/154044288-c8431711-8608-4232-9206-170a335cfbb2.jpg)

**Comment by knelk** [comment]
@manuelmaiorb can you please update your uploaded photo? (it appears fully black in my browser).

did you solve the issue, or does it persist?

**Reply by manuelmaiorb** [reply]
@knelk 
I have managed to find the potential problem. It seems that I had multiplied the normalized Optical Flow values with _width_ and _height_ in another code section and the color values got saturated and caused the staircase effect.

**Reply by JinghangLi** [reply]
@manuelmaiorb  Hello, I also encountered "ladder" when getting optical flow data, can you please show me the complete code you used to solve this problem? From np.frombuffer(） to flow_map, thank you very much for your help.
